src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -115,28 +115,59 @@ protected override void ApplyResponseChallenge()
         /// Responds to a 401 Challenge. Sends an OpenIdConnect message to the 'identity authority' to obtain an identity.
         /// </summary>
         /// <returns></returns>
+        /// <remarks>Uses log id's OIDCH-0026 - OIDCH-0050, next num: 37</remarks>
         protected override async Task ApplyResponseChallengeAsync()
         {
+            bool logLevelError = _logger.IsEnabled(LogLevel.Error);
+            bool logLevelDebug = _logger.IsEnabled(LogLevel.Debug);
+            bool logLevelInformational = _logger.IsEnabled(LogLevel.Information);
+            bool logLevelWarning = _logger.IsEnabled(LogLevel.Warning);
+
+            if (logLevelDebug)
+            {
+                if (logLevelDebug)
+                {
+                    _logger.LogDebug(string.Format(CultureInfo.CurrentCulture, Resources.OIDCH_0026_ApplyResponseChallengeAsync, this.GetType().ToString()));
+                }
+            }
+
             if (ShouldConvertChallengeToForbidden())
             {
+                if (logLevelDebug)
```

Comment:
I always do measure before speaking :-). Following are the results of checking a bool vs. calling into a method. I had to bump it up so high because at 500,000 I was getting '0' for the 'bool' check.

Checking bools: '00:00:00.0155706'
Iterations: '1000000'
Calling method: '00:00:01.1094496'
Iterations: '1000000'

That said, I will back it out if everyone thinks it's not worth it.